### PR TITLE
X sp user profile property

### PR DESCRIPTION
**Pull request by camiloborges**
massive module. I've included commented out lines suggesting function break down.

User Profile Sections are not covered on this one.

### CODE REVIEW

File: Modules/xSharePoint/DSCResources/MSFT_xSPUserProfileProperty/MSFT_xSPUserProfileProperty.psm1 [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+            return $null
+        }
+        
+        $caURL = (Get-SpWebApplication  -IncludeCentralAdministration | ?{$_.IsAdministrationWebApplication -eq $true }).Url
+        $context = Get-SPServiceContext  $caURL 
+
+        $userProfileConfigManager = new-object Microsoft.Office.Server.UserProfiles.UserProfileConfigManager($context)
+        if($null -eq $userProfileConfigManager)
+        {   #if config manager returns when ups is available then isuee is permissions
+            throw "account running process needs permissions"
```
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPUserProfileProperty/MSFT_xSPUserProfileProperty.psm1]: Can we be more specific with the "permissions" that are required here? Just so the error is more meaningful. What are the permissions anyway? Are they permissions we can grant and then revoke when we are done if needed?

**Comment by BrianFarnhill** [comment]
@camiloborges Dude that is a monster bit of code, but it all looks pretty solid to me. I'll kick the tyres in some integration testing closer to releasing it, but again can I get you to do the readme file update as well mate.

**Comment by camiloborges** [comment]
@BrianFarnhill hey, it seems I will have to do some work here to make this mergeable. 
I am working on it

**Comment by camiloborges** [comment]
Hey @BrianFarnhill it seems I will have to close this PR and merge from PowerShell/dev.

**Comment by BrianFarnhill** [comment]
Why close it mate? Ping me a message but you could just pull from the dev branch and then resolve the conflict?

**Pull request closed without merging**